### ISSUE TITLE
Defend against crate feature-variant breaking API changes in CI

### ISSUE THREAD

**Issue by ctz** [rustls/rustls#2190]
In [0.23.14](https://github.com/rustls/rustls/releases/tag/v%2F0.23.14) we had an accepted breaking change due to the signature of [CryptoProvider::install_default()](https://docs.rs/rustls/latest/rustls/crypto/struct.CryptoProvider.html#method.install_default) varying with the `std` crate feature.

I just came across another instance of this, where:

- `ticketer::TicketSwitcher::new`
- `crypto::ring::ticketer::Ticketer::new`
- `crypto::aws_lc_rs::ticketer::Ticketer::new`
 
without the `std` feature all gain an extra argument plus a type parameter.

https://github.com/obi1kenobi/cargo-semver-checks/issues/538#issuecomment-1734225318 contains a hint how this might be possible to check in CI.

**Comment by JakkuSakura** [comment]
This also breaks binaries newly built. I need to add some commands manually to set per-process crypto provider

**Comment by cpu** [comment]
> This also breaks binaries newly built. I need to add some commands manually to set per-process crypto provider

I don't think this is related to the topic of this issue. It sounds like you're referring to something like https://github.com/rustls/rustls/issues/2148, which was called out as a major change in [0.23.0](https://github.com/rustls/rustls/releases/tag/v%2F0.23.0)'s release notes. Please open a separate issue if you're having trouble.
